src/participant.rs

```diff
@@ -72,6 +72,31 @@ pub(crate) trait ProtocolParticipant {
         Ok(messages)
     }
 
+    fn fetch_messages_by_sender(
+        &mut self,
+        message_type: MessageType,
+        sid: Identifier,
+        sender: ParticipantIdentifier,
```

Comment:
I just pushed some code which turns the repeated code into calls to `get_message_queue` and `write_message_queue`. However, these functions are now a part of the `ProtocolParticipant` trait, but ideally wouldn't show up there. Any thoughts on how to handle this?

Reply:
I think this is a @gatoWololo question, but since the trait itself is `pub(crate)` and folks outside the library won't see it, I think this is an okay solution.